mobile/test/swift/integration/EndToEndNetworkingTest.swift

```diff
@@ -1,28 +1,26 @@
 import Envoy
+import EnvoyTestServerInterface
+import Foundation
 import TestExtensions
 import XCTest
 
-final class DirectResponseExactPathMatchIntegrationTest: XCTestCase {
+final class EndToEndNetworkingTest: XCTestCase {
   override static func setUp() {
     super.setUp()
     register_test_extensions()
   }
 
-  func testDirectResponseWithExactPathMatch() {
+  func testNetworkRequestReturnsHeadersAndData() {
+    TestServerInterface.startHttpTestServer()
     let headersExpectation = self.expectation(description: "Response headers received")
     let dataExpectation = self.expectation(description: "Response data received")
-
+    let port = String(TestServerInterface.getServerPort())
     let requestHeaders = RequestHeadersBuilder(
-      method: .get, authority: "127.0.0.1", path: "/v1/abc"
-    ).build()
+      method: .get, scheme: "http", authority: "localhost:" + port, path: "/"
+    )
+    .build()
 
-    let engine = TestEngineBuilder()
-      .addDirectResponse(
-        .init(
-          matcher: RouteMatcher(fullPath: "/v1/abc"),
-          status: 200, body: "hello world", headers: ["x-response-foo": "aaa"]
-        )
-      )
```

Comment:
I think it would be good to keep this in this test file so that the expectations on line 39 have context.